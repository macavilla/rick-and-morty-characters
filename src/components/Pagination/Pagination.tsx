import Link from "next/link";

import getParams from "@/utils/getParams";

import { PaginationType } from "@/types/pagination";

export interface PaginationProps {
  paginationInfo: PaginationType;
}

const Pagination = ({ paginationInfo }: PaginationProps) => {
  const { prev, next, count, pages } = paginationInfo || {};

  const params =
    prev || next
      ? {
          prev: prev ? getParams(prev) : "",
          next: next ? getParams(next) : "",
        }
      : {};
  return (
    <nav className="fixed bottom-1/2 w-full left-0 z-10">
      {params && (
        <ul className="w-full flex px-4">
          {params.prev && (
            <li className="mr-auto size-10 rounded-full bg-indigo-500 p-2 flex justify-center items-center">
              <Link aria-description="Go to previous page" href={params.prev}>
                {"<<"}
              </Link>
            </li>
          )}
          {params.next && (
            <li className="ml-auto size-10 rounded-full bg-indigo-500 p-2 flex justify-center items-center">
              <Link aria-description="Go to next page" href={params.next}>
                {">>"}
              </Link>
            </li>
          )}
        </ul>
      )}
    </nav>
  );
};

export default Pagination;
