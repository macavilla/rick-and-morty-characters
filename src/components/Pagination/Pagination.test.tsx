import { screen, render, waitFor } from "@testing-library/react";
import { mockPaginationProps } from "./__mockData__/Pagination.mock";

import Pagination from "./Pagination";

describe("Pagination", () => {
  it("renders Pagination and matches snapshot", async () => {
    const { container } = render(<Pagination {...mockPaginationProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should render Pagination both arrows ", async () => {
    render(<Pagination {...mockPaginationProps} />);

    screen.findByRole("link", { name: "<<" });
    screen.findByRole("link", { name: ">>" });
  });

  it("renders only prev link", () => {
    render(
      <Pagination
        paginationInfo={{
          ...mockPaginationProps.paginationInfo,
          next: null,
        }}
      />
    );
    expect(screen.getByText("<<")).toBeInTheDocument();
    expect(screen.queryByText(">>")).not.toBeInTheDocument();
  });

  it("renders only next link", () => {
    render(
      <Pagination
        paginationInfo={{
          ...mockPaginationProps.paginationInfo,
          prev: null,
        }}
      />
    );
    expect(screen.getByText(">>")).toBeInTheDocument();
    expect(screen.queryByText("<<")).not.toBeInTheDocument();
  });
});
