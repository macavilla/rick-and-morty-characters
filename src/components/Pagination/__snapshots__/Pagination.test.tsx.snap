// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pagination renders Pagination and matches snapshot 1`] = `
<div>
  <nav
    class="fixed bottom-1/2 w-full left-0 z-10"
  >
    <ul
      class="w-full flex px-2"
    >
      <li
        class="mr-auto text-xl transition duration-300 ease-in-out hover:bg-indigo-800 size-10 rounded-full bg-indigo-500 p-2 flex justify-center items-center"
      >
        <a
          aria-description="Go to previous page"
          href="?page=3"
        >
          &lt;&lt;
        </a>
      </li>
      <li
        class="ml-auto text-xl transition duration-300 ease-in-out hover:bg-indigo-800 size-10 rounded-full bg-indigo-500 p-2 flex justify-center items-center"
      >
        <a
          aria-description="Go to next page"
          href="?page=2"
        >
          &gt;&gt;
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
